src/main/java/org/jabref/logic/importer/fetcher/CollectionOfComputerScienceBibliographiesParser.java

```diff
@@ -0,0 +1,101 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.importer.Parser;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+public class CollectionOfComputerScienceBibliographiesParser implements Parser {
+    @Override
+    public List<BibEntry> parseEntries(InputStream inputStream) throws ParseException {
+        try {
+            Document document = buildDocumentFromInputStream(inputStream);
+            // uncomment to generate test case xml
+            // XMLUtil.printDocument(document);
+
+            NodeList childNodes = document.getChildNodes();
+            List<Element> itemElements = findItemElementsRecursively(childNodes);
+            List<BibEntry> bibEntries = parseItemElements(itemElements);
+
+            // uncomment to generate test case bib files
+            // System.out.println(bibEntries);
+
+            return bibEntries;
+        } catch (ParserConfigurationException | SAXException | IOException exception) {
+            throw new ParseException(exception);
+        }
+    }
+
+    private Document buildDocumentFromInputStream(InputStream inputStream) throws ParserConfigurationException, SAXException, IOException {
+        DocumentBuilder dbuild = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+        Reader reader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
```

Comment:
Not sure whether this should be closed, too.

I think, however, that closing the `inputStream` is enough. (Which surely happens somehwere in the code)

Reply:
Good point. Actually, unless I'm missing something, I don't think the inputStream is closed - I will refactor this code so that I make sure to shut the inputStream and the reader in a finally block.

Reply:
Typically, one should use the try-with-resoruces feature by Java --> https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Could be hard here since the thing is distributed among multiple methods

Reply:
Actually you are correct - they were using try-with-resources for the InputStream. I have used the same pattern for the InputStream I create in my latest commit.